### PR TITLE
app: land Tom's Cody-in-system-tray experiment

### PR DESCRIPTION
**Pull request by slimsag**
I've revamped Tom's PR https://github.com/sourcegraph/sourcegraph/pull/51266 and updated the branch since there were quite a few conflicts that were a bit tedious to solve. I also fixed a good number of bugs.

I think it would be worth merging this as-is, once CI passes, to avoid more merge conflicts and then we can continue to iterate on it.

## Test plan

`sg start app` -> manually test the behavior from system tray

### CODE REVIEW

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.01% (+0.28 kb) | 0.10% (+14.19 kb) 🔺 | 0.12% (+13.91 kb) 🔺 | 0.13% (+1) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/sg/system-tray-cody/compare-report.html;tab=live_object) for a full comparison between the commits 4310023 and 2d81654 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by marekweb** [comment]
Nice! 

Works in `sg start app`. The release build doesn't start though, but I don't see how this PR would cause that so maybe it's an unrelated issue?

```
[2023-05-11][12:32:56][ERROR][app] ✱ Sourcegraph App version: 5.0.2
[2023-05-11][12:33:27][ERROR][app] unable to set up PostgreSQL: Failed to download or start embedded postgresql. Please start your own postgres instance then configure the PG* environment variables to connect to it as well as setting USE_EMBEDDED_POSTGRESQL=0: timed out waiting for database to become available
```

**Review by marekweb — APPROVED** [review]

File: client/web/src/routes.tsx [hunk 1]
```diff
@@ -155,6 +156,10 @@ export const routes: RouteObject[] = [
         // expose this information in the handle object instead.
         handle: { isRepoContainer: true },
     },
+    {
+        path: PageRoutes.CodyStandalone,
+        element: <CodyStandalonePage />,
+    },
```
Comment (limitedmage) [inline client/web/src/routes.tsx:162]: This route should only be available in the Sourcegraph app, can we add a condition here?

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -62,11 +75,15 @@ fn main() {
     tauri::Builder::default()
         .system_tray(tray)
         .on_system_tray_event(tray::on_system_tray_event)
+        .on_system_tray_event(|app, event| {
+            tauri_plugin_positioner::on_tray_event(app, &event);
+        })
         .on_window_event(|event| match event.event() {
             tauri::WindowEvent::CloseRequested { api, .. } => {
-                // Ensure the app stays open after the last window is closed.
```
Comment (limitedmage) [inline src-tauri/src/main.rs:67]: nit: don't remove this comment

**Review by limitedmage — APPROVED** [review]

File: client/web/src/routes.tsx (hunk 1 above)
Comment (slimsag) [reply client/web/src/routes.tsx]: I struggled quite a bit trying to figure out how to do this correctly; I arrived at a solution that works but I'm afraid it's not the best / right way to do things here; and Cody was useless for this; could you take a look and/or give me some guidance here?

Comment (limitedmage) [reply client/web/src/routes.tsx]: What we do in other pages is [wrap in `LegacyRoute`](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Asourcegraph%2Fsourcegraph.git&branch=main&file=client%2Fweb%2Fsrc%2Fenterprise%2Froutes.tsx&editor=VSCode&version=2.2.16&start_row=112&start_col=0&end_row=115&end_col=10)

**Pull request merged**